src/hooks/useTrackedState.js

```diff
@@ -0,0 +1,92 @@
+/* eslint-env es6 */
+
+import { useReducer, useRef, useMemo, useContext } from 'react'
+import { useReduxContext as useDefaultReduxContext } from './useReduxContext'
+import Subscription from '../utils/Subscription'
+import { useIsomorphicLayoutEffect } from '../utils/useIsomorphicLayoutEffect'
+import { ReactReduxContext } from '../components/Context'
+import { createDeepProxy, isDeepChanged } from '../utils/deepProxy'
+
+function useTrackedStateWithStoreAndSubscription(store, contextSub) {
+  const [, forceRender] = useReducer(s => s + 1, 0)
+
+  const subscription = useMemo(() => new Subscription(store, contextSub), [
+    store,
+    contextSub
+  ])
+
+  const state = store.getState()
+  const affected = new WeakMap()
+  const latestTracked = useRef(null)
+  useIsomorphicLayoutEffect(() => {
+    latestTracked.current = {
+      state,
+      affected,
+      cache: new WeakMap()
+    }
+  })
+  useIsomorphicLayoutEffect(() => {
```

Comment:
While it's not an immediate concern, I'm noticing that we now have 3 different parts of our codebase where we're subscribing to the store in a hook.  That _may_ be worth looking at to see if there's some way to abstract it.